### ISSUE TITLE
Frequent false alerts from Monitor Kubecost Health

### ISSUE THREAD

**Issue by praveenkumar0566** [kubecost/cost-analyzer-helm-chart#1817]
Thanks for your interest in Kubecost!  To ensure you get the help you need, please follow these guidelines:

# For reproducible bugs, please provide the following
**Describe the bug**

After configuring "Monitor Kubecost Health" to notify a slack channel. There are false alert notifications to slack "Added cluster" and "removed cluster" in frequent intervals. One of the customer's issues
They have 8 clusters set up and they are facing this issue with 1 of them. 

**To Reproduce**

1. Go to 'Alerts on a primary cluster
1. Click on 'Monitor Kubecost Health'
1. configure the details
1. If there are a couple of secondary clusters, one of the clusters sends a false alarm "Added cluster" "removed cluster"

**Expected behavior**

False alerts should not be sent.

**Screenshots**

![](https://user-images.githubusercontent.com/102574445/203368091-db0c0fe1-91d9-40be-b499-f0fc631decf8.png)

![](https://user-images.githubusercontent.com/102574445/203368142-04f84d9b-4f35-455a-adfb-479f9da73284.png)

┆Issue is synchronized with this [Jira Task](https://kubecost.atlassian.net/browse/GIB-215) by [Unito](https://www.unito.io)

**Comment by AjayTripathy** [comment]
First need to rule out that the data upload process is not flaking on the user's end. I've followed up with the user but @praveenkumar0566 can you please get this added to the runbook for debugging this alert?

1. Log into Thanos: 
```kubectl port-forward svc/kubecost-thanos-query-http 8080:10902 --namespace kubecost```
2. Visit localhost:8080
3. Click on the "graph" section
4. Graph ```kubecost_cluster_info``` over 24h or so
5. Send a screenshot of the results to support.

**Comment by dwbrown2** [comment]
@AjayTripathy agree this is valuable to confirm, but even if it is... does sending an alert every 10m feel right?

**Comment by praveenkumar0566** [comment]
> First need to rule out that the data upload process is not flaking on the user's end. I've followed up with the user but @praveenkumar0566 can you please get this added to the runbook for debugging this alert?
> 
> 1. Log into Thanos:
>    `kubectl port-forward svc/kubecost-thanos-query-http 8080:10902 --namespace kubecost`
> 2. Visit localhost:8080
> 3. Click on the "graph" section
> 4. Graph `kubecost_cluster_info` over 24h or so
> 5. Send a screenshot of the results to support.

Sure, I will add it to the runbook.

**Comment by AjayTripathy** [comment]
@dwbrown2 I agree. Here's a PR to support a number of retries https://github.com/kubecost/kubecost-cost-model/pull/1084

**Comment by mbolt35** [comment]
@dwbrown2 This does appear to only send an alert if the status changes, so the fact that it's showing up, then dropping, then showing up again, repeatedly seems to be indicative of something bad happening. In this case, I think the alert is doing exactly what we intended (despite it being delayed 3h anyways). 

What @AjayTripathy added is a one-hour down time requirement (after the metrics have been delayed for 3h) before we alert. I think this will definitely make this specific scenario less spammy, but I tend to look at the spammy alerts as a sign we're doing an adequate job of communicating a potential issue. With the proposed changes, this behavior (up, down, toggling) would be concealed without ever issuing a single alert.

**Comment by mbolt35** [comment]
@AjayTripathy I think we should actually change the way the cluster map works maybe? I think it issues instant vector queries for the `kubecost_cluster_info` metric. Should we be issuing something like `avg_over_time(kubecost_cluster_info[10m])`? 

https://github.com/opencost/opencost/blob/a34125a484b0aecf6379738ee3204dfcf02e76ab/pkg/costmodel/clusters/clustermap.go#L132-L135

I've discussed this with a few folks in the last few weeks, and I do think based on the number of services dependent on the `ClusterMap`, it makes sense to be a bit more resilient to missed scrapes. 10m feels like an adequate window of time to raise an alert for a missing cluster? Seems reasonable that we could make cluster map a bit more resilient, and you could leave avoid making changes to the alert logic? 

Let me know what you think -- I think I'd prefer the change to ClusterMap (since it would address a few other bugs as well).

**Comment by AjayTripathy** [comment]
Sure, sgtm. Want to take a quick PR on that then and I'll update mine?
